### ISSUE TITLE
Update Managed Identity documentation with Linux Consumption limitations

### ISSUE THREAD

**Issue by bachuv** [Azure/azure-functions-durable-extension#2525]
Opening this issue to track updating the Durable Functions documentation to warn customers that Managed Identity doesn't work with Linux Consumption in certain scenarios.

@davidmrdavid can you provide more information about the limitations and how they can be reproduced? Thanks!

**Comment by davidmrdavid** [comment]
Yes. The steps are simply to follow our MI quickstart: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-configure-durable-functions-with-credentials#configure-your-app-to-use-managed-identity-recommended but deploy it (using VSCode) to linux consumption.

You should find two things:
(1) New deployments will fail
(2) The functions runtime may become unavailable.

The issue appears to be that Linux Consumption doesn't support having the default AzureWebJobsStorage connection behind MI. When this is the case, it fails to find the **zip-deployed** code and therefore fails to start the app.

@nytian: since you led the Managed Identity effort, any chance you could try this out?
